src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala

```diff
@@ -40,7 +40,7 @@ trait TestUtils extends Logging {
   }
 
   def schemaEqual(df1: DataFrame, df2: DataFrame): Boolean = {
-    df1.schema.toDDL == df2.schema.toDDL // ignore nullability in comparision
+    df1.schema.sql == df2.schema.sql // ignore nullability in comparison
```

Comment:
```suggestion
  def schemaEqual(df1: DataFrame, df2: DataFrame): Boolean = {
    def makeFieldNullable(sf: StructField): StructField = StructField(sf.name, sf.dataType, true, sf.metadata)
    df1.schema.map(makeFieldNullable) == df2.schema.map(makeFieldNullable) // ignore nullability in comparison
  }
```
I would rather prefer a typed comparison even if the code is longer.

Reply:
Also optimise imports please, i.e. remove
```
import java.sql.Timestamp
import org.apache.spark.sql.types.{DataType, DataTypes, StructType}
```

Reply:
@kaelte thanks for the suggestion, but your `def makeFieldNullable` just compares the top level attributes. There might be nested structs where we need to ignore nullability and metadata as well. I would stay with the short version `.sql` which can handle these cases, as it is only for testing.

Reply:
Imports optimized.

Reply:
> `makeFieldNullable`

Yes, you are right. We can stick with `.sql` or use the same method as in SDL:
https://github.com/smart-data-lake/smart-data-lake/blob/develop-spark3/sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala#L74